docs/java/how-to-develop-a-workflow-definition-in-java.md

```diff
@@ -0,0 +1,186 @@
+title: How to develop a Workflow Definition in Java
+sidebar_label: Workflow Definition
+description: In the Temporal Java SDK programming model, a Workflow is a class which implements a Workflow Interface.
+---
+
+import RelatedReadList from '../components/RelatedReadList.js'
+
+In the Temporal Java SDK programming model, a Workflow is a class which implements a Workflow Interface:
+
+```java
+public class FileProcessingWorkflowImpl implements FileProcessingWorkflow {
+  // ...
+}
+```
+
+## Workflow Interface
+
+The Workflow Interface is a Java interface and is annotated with `@WorkflowInterface`. Workflow interface methods must have one of the following annotations:
+
+- **@WorkflowMethod** denotes the starting point of Workflow Execution. Workflow Execution completes when this methods returns.
+- **@SignalMethod** indicates that this method is a Signal handler method and that it can react to external Signals. It can have parameters which can contain the Signal payload. It does not return a value, so it must have a `void` return type.
+- **@QueryMethod** indicates that this method can be used to Query the Workflow's state at any time during its execution.
+  It can have parameters which can be used to filter a subset of the Workflow's state that it returns. Since it does return a value it must have a non `void` return type.
+
+Workflow interfaces can define only a single method annotated with `@WorkflowMethod`. They can define
+any number of methods annotated with `@SignalMethod` and `@QueryMethod`, for example:
+
+```java
+@WorkflowInterface
+public interface FileProcessingWorkflow {
+
+    @WorkflowMethod
+    String processFile(Arguments args);
+
+    @QueryMethod(name="history")
+    List<String> getHistory();
+
+    @QueryMethod
+    String getStatus();
+
+    @SignalMethod
+    void retryNow();
+
+    @SignalMethod
+    void abandon();
+}
+```
+
+The `@WorkflowMethod` annotation has a `name` parameter, for example: `@WorkflowMethod(name = "MyWorkflowType")`.
+It can be used to denote the Workflow type. If not set, the Workflow type defaults to the short name of the Workflow interface. In the example above, the Workflow type defaults to `FileProcessingWorkflow`.
+Methods annotated with `@WorkflowMethod` can have any number of parameters. We recommend passing a single parameter that contains all the input fields. This allows adding fields in a backward compatible manner.
+
+The `@QueryMethod` annotation also has a `name` parameter, for example: `@QueryMethod(name = "history")`. It can be
+used to denote the Query name. If not set, the Query name defaults to the name of the method. In the example above, the Query name defaults to `getStatus`.
+
+The `@SignalMethod` too has a `name` parameter, for example: `@SignalMethod(name = "mysignal")`. It can be used to denote the
+Signal type. If not set, the Signal type defaults to the name of the method. In the example above, the Signal type defaults to `retryNow` and `abandon`.
```

Comment:
```suggestion
It can be used to denote the Signal type.
If not set, the Signal type defaults to the name of the method.
In the preceding example, the Signal type defaults to `retryNow` and `abandon`.
```